src/Microsoft.AspNetCore.Mvc.Razor.Precompilation.Design/Internal/PrecompileRunCommand.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNetCore.Mvc.Razor.Compilation;
+using Microsoft.AspNetCore.Razor.CodeGenerators;
+using Microsoft.Cci;
+using Microsoft.Cci.MutableCodeModel;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.Extensions.CommandLineUtils;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Precompilation.Design.Internal
+{
+    public class PrecompileRunCommand
+    {
+        private readonly string PrecompiledAssemblyNameSuffix = Guid.NewGuid().ToString();
+
+        private CommandOption OutputFilePath { get; set; }
+
+        private MvcServicesProvider ServicesProvider { get; set; }
+
+        private CommonOptions Options { get; } = new CommonOptions();
+
+        private string ProjectPath { get; set; }
+
+        public void Configure(CommandLineApplication app)
+        {
+            Options.Configure(app);
+            OutputFilePath = app.Option(
+                "--output-file-path", 
+                "Path to the output binary (assembly or executable).",
+                CommandOptionType.SingleValue);
+
+            app.OnExecute(() => Execute());
+        }
+
+        private int Execute()
+        {
+            ParseArguments();
+
+            ServicesProvider = new MvcServicesProvider(
+                ProjectPath,
+                OutputFilePath.Value(),
+                Options.ContentRootOption.Value(),
+                Options.ConfigureCompilationType.Value());
+
+            Console.WriteLine("Running Razor view precompilation.");
+
+            var result = CompileViews();
+            if (!result.Success)
+            {
+                foreach (var error in result.RazorErrors)
+                {
+                    Console.Error.WriteLine($"{error.Location.FilePath} ({error.Location.LineIndex}): {error.Message}");
+                }
+
+                foreach (var error in result.RoslynErrors)
+                {
+                    Console.Error.WriteLine(CSharpDiagnosticFormatter.Instance.Format(error));
+                }
+
+                return 1;
+            }
+
+            Console.WriteLine($"Successfully compiled {result.CompileOutputs.Count} Razor views.");
+            Console.WriteLine($"Injecting precompiled views into assembly {OutputFilePath.Value()}.");
+
+            UpdateAssembly(result);
+            return 0;
+        }
+
+        private void ParseArguments()
+        {
+            ProjectPath = Options.ProjectArgument.Value;
+            if (string.IsNullOrEmpty(ProjectPath))
+            {
+                throw new ArgumentException("Project path not specified.");
+            }
+
+            if (!OutputFilePath.HasValue())
```

Comment:
Why make this required? Can't we just use the default project/bin location if this isn't specified?